src/commands/captureDiagnostics.ts

```diff
@@ -0,0 +1,283 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2021-2024 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as fs from "fs/promises";
+import * as path from "path";
+import * as vscode from "vscode";
+import { tmpdir } from "os";
+import { exec } from "child_process";
+import { Writable } from "stream";
+import { WorkspaceContext } from "../WorkspaceContext";
+import { Version } from "../utilities/version";
+import { execFileStreamOutput } from "../utilities/utilities";
+import configuration from "../configuration";
+
+export async function captureDiagnostics(
+    ctx: WorkspaceContext,
+    allowMinimalCapture: boolean = true
+) {
+    try {
+        const captureMode = await captureDiagnosticsMode(ctx, allowMinimalCapture);
+
+        // dialog was cancelled
+        if (!captureMode) {
+            return;
+        }
+
+        const diagnosticsDir = path.join(
+            tmpdir(),
+            `vscode-diagnostics-${formatDateString(new Date())}`
+        );
+
+        await fs.mkdir(diagnosticsDir);
+        await writeLogFile(diagnosticsDir, "extension-logs.txt", extensionLogs(ctx));
+        await writeLogFile(diagnosticsDir, "settings.txt", settingsLogs(ctx));
+        await writeLogFile(diagnosticsDir, "source-code-diagnostics.txt", diagnosticLogs());
```

Comment:
I think we should only add the source code diagnostics in the full capture mode because it may contain information about the user’s source code.

Reply:
Done